blog/2020-09-25-gh-release.md

```diff
@@ -0,0 +1,165 @@
+
+> This will create a Release, as well as a release event, which could be handled by a third party service, or by GitHub Actions for additional uses, for example the @actions/upload-release-asset GitHub Action.
+
+Eureka!
+By splitting my workflow into two, I no longer needed to worry about the above problem.
+I would simply have [one workflow](https://github.com/comit-network/comit-rs/blob/dev/.github/workflows/create-cnd-gh-release.yml) that creates the release and is only run on a single platform and a [second workflow](https://github.com/comit-network/comit-rs/blob/dev/.github/workflows/release-cnd.yml) that is triggered by the `release` event and runs on all platforms we want to build our binary on.
+The `release` event contains the `upload_url` so we can simply pass it into the `@actions/upload-release-asset` action.
+
+## Local actions for code-reuse
+
+Whilst having modular workflows was a good experience, it did create some code duplication between the different workflows.
+With GitHub actions, the unit of code reuse is a GitHub action.
+However, releasing a GitHub action to the marketplace felt like an overkill for this problem.
+Fortunately, a step can [reference](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-using-action-in-the-same-repository-as-the-workflow) an action that is stored within the same repository.
+
+This allowed me to extract [two](https://github.com/comit-network/comit-rs/tree/dev/.github/actions) otherwise duplicated sections of different workflows into dedicated actions and reuse them.
+
+## Releasing several components from the same repository
+
+The `comit-rs` repository is a Cargo workspace and therefore home to several crates.
+Two of them are binaries that are released using the workflows discussed in this post.
+In the future, we might also release crates to https://crates.io/ from the repository.
+
+Especially with the two binaries, I had to make a decision in the workflow design:
+
+The release process is very similar, do I make one configurable workflow or two very similar ones?
+
+In the end, I settled with two separate workflows that are almost identical.
+While one could complain about this duplicated code, I don't think it is very harmful in this case.
+
+1. The two workflows are similar for now but that doesn't mean they always will be.
+There is already one difference: `cnd` is also published as a docker image, whereas `nectar` isn't.
+It is little details like that which would require configuration of the workflow, making it harder to maintain.
+2. Testing workflows is mostly a manual process.
+Not having to worry about different configuration combinations greatly simplifies maintenance.
+3. It is much easier to follow along what the workflow does if there are several "hardcoded" values like [the prefix of the tag](https://github.com/comit-network/comit-rs/blob/acabab36058b4e3d3ae54bde10fd7ce5db66f1a5/.github/workflows/release-cnd.yml#L32-L37).
+
```

Comment:
You could mention the motto " copy once, refactor on the third go" or whatever the motto is :)